### PR TITLE
FJS-1411: fix displaying server errors

### PR DESCRIPTION
**Pull request by DmitryNistratov**

### CODE REVIEW

**Review by TanyaGashtold — APPROVED** [review]

File: src/Webform.js [hunk 1]
```diff
@@ -1474,11 +1486,24 @@ export default class Webform extends NestedDataComponent {
   executeSubmit(options) {
     this.submitted = true;
     this.submitting = true;
+    this.clearServerErrors();
```
Comment (alexandraRamanenka) [inline src/Webform.js:1489]: Why did you place clearServerErrors in executeSubmit and not inside submit? Isn't it possible that beforeSubmit() throws an Error and then executeSubmit won't be executed?

**Review by alexandraRamanenka — APPROVED** [review]

File: src/Webform.js (hunk 1 above)
Comment (DmitryNistratov) [reply src/Webform.js:1489]: If I understand correctly, server errors should only be cleaned up when we have sent submission on the server. We will then receive a response that will either return updated errors or no errors at all. In either case, you need to clean up old mistakes

**Comment by travist** [comment]
Needs to be rebased.

File: src/Webform.js [hunk 2]
```diff
@@ -1266,6 +1274,10 @@ export default class Webform extends NestedDataComponent {
       if ('details' in error) {
         error = error.details;
       }
+      this.serverErrors = error.map((err) => {
+        err.fromServer = true;
```
Comment (travist) [inline src/Webform.js:1278]: why are you setting ALL the errors to be fromServer here?  How do we know they are from the server?

**Review by travist — CHANGES_REQUESTED** [review]

**Pull request merged**